schema/hardware-metadata.schema.json

```diff
@@ -255,6 +277,15 @@
         },
         "exposes": {
           "$ref": "#/$defs/interconnects"
+        },
+        "revisions": {
```

Comment:
Shield *variants* are sadly not nearly as structured as board revisions. See https://docs.zephyrproject.org/3.5.0/hardware/porting/shields.html#shield-variants

This probably needs to be handled differently, or just not handled at all in this PR, so we can focus on getting board revisions in first.